### PR TITLE
Add support for testing whether a value is a string in Pascal case

### LINKED ISSUE #536: [RFC]: add support for testing whether a value is a Pascal-cased string
### Description

This RFC proposes to add support for testing whether a value is a Pascal-cased string.

```javascript
var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
var pascalcase = require( '@stdlib/string/base/pascalcase' );

function isPascalcase( value ) {
    return ( isString( value ) && pascalcase( value ) === value );
}
```

Package: `@stdlib/assert/is-pascalcase`
alias: `isPascalcase`

### Related Issues

No.

### Questions

No.

### Other

Reference packages:

-   [`@stdlib/assert/is-uppercase`](https://github.com/stdlib-js/stdlib/tree/13c1fa39ff73208d819697dc07917bdb7539f295/lib/node_modules/%40stdlib/assert/is-uppercase)

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #536 .

## Description

> What is the purpose of this pull request?

This pull request:

-   adds assertion utility `isPascalCase`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #536 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Planeshifter** [comment]
@kgryte The package is now ready for review.

File: lib/node_modules/@stdlib/assert/is-pascalcase/docs/repl.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    Parameters
+    ----------
+    value: any
+        Value to test.
+
+    Returns
+    -------
+    bool: boolean
+        Boolean indicating whether value is a string in Pascal case.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/docs/repl.txt]:
```suggestion
        Boolean indicating whether a value is a string in Pascal case.
```

File: lib/node_modules/@stdlib/assert/is-pascalcase/docs/types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Tests if a `value` is a `string` in Pascal case.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/docs/types/index.d.ts]:
```suggestion
* Tests if a value is a string in Pascal case.
```

File: lib/node_modules/@stdlib/assert/is-pascalcase/docs/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Tests if a value is a string in Pascal case.
+*
+* @param value - value to test
+* @returns boolean indicating if a value is a `string` in Pascal case
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/docs/types/index.d.ts]:
```suggestion
* @returns boolean indicating if a value is a string in Pascal case
```

File: lib/node_modules/@stdlib/assert/is-pascalcase/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+var pascalcase = require( '@stdlib/string/base/pascalcase' );
+
+
+// MAIN //
+
+/**
+* Tests if a `value` is a `string` in Pascal case.
+*
+* @param {*} value - value to test
+* @returns {boolean} boolean indicating if a value is a `string` in Pascal case
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/lib/main.js]:
```suggestion
* Tests if a value is a string in Pascal case.
*
* @param {*} value - value to test
* @returns {boolean} boolean indicating if a value is a string in Pascal case
```

File: lib/node_modules/@stdlib/assert/is-pascalcase/test/test.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		'!'
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.equal( isPascalcase( values[i] ), false, 'returns false when provided '+values[i] );
+	}
+	t.end();
+});
+
+tape( 'the function returns `false` if not provided a string primitive', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/test/test.js]:
```suggestion
tape( 'the function returns `false` if not provided a string', function test( t ) {
```

File: lib/node_modules/@stdlib/assert/is-pascalcase/test/test.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	}
+	t.end();
+});
+
+tape( 'the function returns `false` if not provided a string', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		new String( 'Beep' ), // eslint-disable-line no-new-wrappers
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/test/test.js]:
```suggestion
```

I think it is fine to omit this test case.

**Review by kgryte — APPROVED** [review]
LGTM

File: lib/node_modules/@stdlib/assert/is-pascalcase/docs/types/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Tests if a value is a string in Pascal case.
+*
+* @param value - value to test
+* @returns boolean indicating if a value is a string in Pascal case
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/docs/types/index.d.ts]:
```suggestion
* @returns boolean indicating whether a value is a string in Pascal case
```

File: lib/node_modules/@stdlib/assert/is-pascalcase/lib/main.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+var pascalcase = require( '@stdlib/string/base/pascalcase' );
+
+
+// MAIN //
+
+/**
+* Tests if a value is a string in Pascal case.
+*
+* @param {*} value - value to test
+* @returns {boolean} boolean indicating if a value is a string in Pascal case
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-pascalcase/lib/main.js]:
```suggestion
* @returns {boolean} boolean indicating whether a value is a string in Pascal case
```

Updating to use consistent language `if/whether`.

**Pull request merged**